lib/TPP/Transforms/TileConfig.cpp

```diff
@@ -0,0 +1,154 @@
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements loop tiling on parallel loops.
+//
+//===----------------------------------------------------------------------===//
+#include "TPP/Dialect/Xsmm/XsmmOps.h"
+#include "TPP/Dialect/Xsmm/XsmmUtils.h"
+#include "TPP/Transforms/Utils/VNNIUtils.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include <iostream>
+#include <list>
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_TILECONFIGINSERTIONPASS
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+using namespace mlir;
+
+namespace mlir {
+namespace tpp {
+
+template <typename InvokeOpTy, typename DispatchOpTy>
+static void appendBrgemmFlags(SmallVector<Attribute> &attributes,
+                              PatternRewriter &rewriter, InvokeOpTy opTy) {
+  auto flags =
+      dyn_cast<DispatchOpTy>(opTy.getOperand(0).getDefiningOp()).getFlags();
+  for (auto flagItr : flags) {
```

Comment:
We have this same pattern twice in the code already (Combine XSMM pass and Linalg to XSMM). 

They're not identical but easily convertible. We can do this in a later PR, but please add a TODO here so that we don't forget, and please do this on a following PR ASAP.